### ISSUE TITLE
Feature: Detect when a something that looks like a variable is used

### ISSUE THREAD

**Issue by adiroiban** [koalaman/shellcheck#961]
You might define a variable like `some_name=something` and if you forget to use it as `$some_name` there should be a warning.

#### For new checks and feature suggestions

- [ x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
some_var=my_value
echo Here we are $some_var
echo This should trigger an error: some_var
```

#### Here's what shellcheck currently says:

Nothing :)

#### Here's what I wanted or expected to see:

I am expecting to complain about line 3 where `some_var` is used as a string, but most of the time you want to call it as `$some_var` to reference its value

**Comment by Pawamoy** [comment]
And what if I use a variable called `success` or `error`? I will get a shellcheck warning each time I write success or error in a string? I would have to disable them everywhere, and it would be very cumbersome :confused:

**Comment by adiroiban** [comment]
> And what if I use a variable called success or error?

good point.

Maybe have this check only in tests , like

```
#!/bin/bash
cache_file=/tmp/cache
echo $cache_file
if [ -f cache_file ]; then
   echo "We have a cache."
fi
```

**Comment by nihilus** [comment]
Yeah, better improve the code itself. ALBEIT this could be make to only test against variable containing a '_'.

**Comment by jleffler** [comment]
One issue could be that the variable is an environment variable, inherited but overridden by the script, and used by programs that the script runs.  There's no way for shell check to know whether the variable would be inherited from the calling environment, or whether it will be used by the programs run by the script.  Thus, detecting this could generate too many false positives.  I'm not entirely sure how plausible this is.  Setting but not using known environment variables could be ignored, but it would be tricky to white-list environment variables relevant to a particular context.

**Comment by brother** [comment]
@adiroiban that example looks like SC2160 - "Instead of '[ true ]', just use 'true'."
